src/genotyper.cc

```diff
@@ -41,6 +41,286 @@ static bool is_pseudo_ref_record(const bcf_hdr_t* hdr, bcf1_t* record) {
+        if (n_val_per_sample < 0) {
+            n_val_per_sample = expected_n_val_per_sample(record);
+        }
+
+        vector<string> names_to_search = field_names;
+        if (names_to_search.empty()) {
+            names_to_search = field_info.orig_names;
+        }
+
+        for (auto& field_name : names_to_search) {
+            if (found) break;
+            T *v = nullptr;
+            int vsz = 0;
+
+            // rv is the number of values written
+            int rv = FormatFieldHelper::bcf_get_format_wrapper(dataset_header, record, field_name.c_str(), &v, &vsz);
+
+            // raise error if there's a failed get due to type mismatch
+            if (rv == -2) {
+                if (v) free(v);
+                ostringstream errmsg;
+                errmsg << dataset << " " << range(record).str() << " (" << field_name << ")";
+                return Status::Invalid("genotyper: getting format field errored with type mismatch", errmsg.str());
+            }
+            // don't raise error if get failed due to tag missing in record or
+            // in vcf header; continue to look with other possible field names
+
+            if (rv >= 0) {
+                found = true;
+                if (rv != record->n_sample * n_val_per_sample) {
+                // For this field, we expect n_val_per_sample values per sample
+                    if (v) free(v);
+                    ostringstream errmsg;
+                    errmsg << dataset << " " << range(record).str() << "(" << field_name << ")";
+                    return Status::Invalid("genotyper: unexpected result when fetching record FORMAT field", errmsg.str());
+                } // close rv != record->n_sample * count
+
+                for (int i=0; i<record->n_sample; i++) {
+                    for (int j=0; j<n_val_per_sample; j++) {
+
+                        int in_ind = i * n_val_per_sample + j;
+                        int out_ind = get_out_ind_of_value(i, j, sample_mapping, allele_mapping);
+
+                        if (out_ind < 0) {
+                           continue;
+                        }
+
+                        assert(out_ind < format_v.size());
+                        assert(in_ind < vsz);
+                        format_v[out_ind].push_back(v[in_ind]);
+                    } // close for j loop
+                } // close for i loop
+            } // close rv >= 0
+            free(v);
+        }
+
+        if (!found) {
+
+            // Special case handling for AD field to convert ref
+            // DP/MIN_DP to repopulate AD field
+            if (field_info.name == "AD" && !handled_AD_field) {
+
+                // Prevent runaway recursion
+                handled_AD_field = true;
+
+                // Call add_record_data again, searching for DP, MIN_DP, override n_val_per_sample to 1
+                Status s = add_record_data(dataset, dataset_header, record, sample_mapping, allele_mapping, {"DP", "MIN_DP"}, 1);
```

Comment:
I think some callers may annotate reference confidence records with both MIN_DP and (average) DP; so we should probably preferentially search MIN_DP first.